app/services/manifest_validator.rb

```diff
@@ -0,0 +1,28 @@
+class ManifestValidator
+  include JsonSchemaHelper
+
+  attr_reader :content, :errors
+
+  def initialize(content)
+    if content.is_a?(String)
+      @content = JSON.parse(content)
+    else
+      @content = content
+    end
+    register_schemas_by_uri
+  end
+
+  def run
+    schema_file_path = schema_file('post-manifest')
+    @errors = JSON::Validator.fully_validate(schema_file_path, content, errors_as_objects: true)
+    !@errors.any?
```

Comment:
what if run returned a hash:

```ruby
{
  errors: true,
  messages: "blah blah"
}
```

then, when using this service class, we'd access both the error and potential messages at the same time. Not sure this implementation is better than what we have here, but something to consider